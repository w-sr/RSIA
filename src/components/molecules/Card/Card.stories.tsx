import { storiesOf } from "@storybook/react";
import { Card } from "./Card";
import { Button } from "../../atoms/Button";

storiesOf("Card", module)
  .add("without button", () => (
    <Card type="FIXED" name="Product" rate={10} lender="lender" width="240px" />
  ))
  .add("with button", () => (
    <Card type="FIXED" name="Product" rate={10} lender="lender" width="240px">
      <Button>Select product</Button>
    </Card>
  ));
