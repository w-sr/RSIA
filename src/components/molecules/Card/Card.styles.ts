import styled, { css } from "styled-components";

export const Card = styled.div<{ width?: string }>`
  ${({ theme, width }) => css`
    width: ${width || "auto"};
    height: 350px;
    padding: 32px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    background-color: ${theme.colors.transparentWhite};
    border-radius: 3px;
    border: ${"1px solid " + theme.colors.black};
    outline: none;
    box-sizing: border-box;
  `}
`;

export const TitleWrap = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Title = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.title.size};
    font-weight: ${theme.font.title.weight};
    line-height: ${theme.font.title.lineHeight};
    text-align: center;
    text-transform: capitalize;
  `}
`;

export const Name = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.medium.size};
    font-weight: ${theme.font.medium.weight};
    line-height: ${theme.font.medium.lineHeight};
    text-align: center;
    margin-top: 8px;
  `}
`;

export const Rate = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.large.size};
    font-weight: ${theme.font.large.weight};
    line-height: ${theme.font.large.lineHeight};
    text-transform: capitalize;
  `}
`;

export const Lender = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.small.size};
    font-weight: bold;
    line-height: ${theme.font.small.lineHeight};
  `}
`;

export const Children = styled.div`
  display: flex;
  justify-content: center;
`;
