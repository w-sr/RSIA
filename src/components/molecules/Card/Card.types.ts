import { ReactNode } from "react";

export enum CardEnum {
  "VARIABLE" = "Variable",
  "FIXED" = "Fix",
}
export type CardProps = {
  type: "VARIABLE" | "FIXED";
  name: string;
  rate: number;
  lender: string;
  width?: string;
  children?: ReactNode;
};
