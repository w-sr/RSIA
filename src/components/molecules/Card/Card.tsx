import * as S from "./Card.styles";
import { CardProps, CardEnum } from "./Card.types";

const Card = ({ name, type, rate, lender, width, children }: CardProps) => (
  <S.Card width={width}>
    <S.TitleWrap>
      <S.Title>{"Best " + CardEnum[type]}</S.Title>
      <S.Name>{"(" + name + ")"}</S.Name>
    </S.TitleWrap>
    <S.Rate>{rate + "%"}</S.Rate>
    <S.Lender>{lender}</S.Lender>
    {children && <S.Children>{children}</S.Children>}
  </S.Card>
);

Card.displayName = "Card";

export { Card };
