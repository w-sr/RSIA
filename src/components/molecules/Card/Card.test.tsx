import { renderWithTheme } from "../../../utils/tests/helpers";
import { Card } from "./Card";

describe("<Card />", () => {
  it("should match the snapshot", () => {
    const { container } = renderWithTheme(
      <Card type="FIXED" name="product" rate={10} lender="lender" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
