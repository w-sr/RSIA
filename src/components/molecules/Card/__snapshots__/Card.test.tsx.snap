// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match the snapshot 1`] = `
<div
  class="sc-bczRLJ cvCJkV"
>
  <div
    class="sc-gsnTZi Msmow"
  >
    <span
      class="sc-dkzDqf eNRxTm"
    >
      Best Fix
    </span>
    <span
      class="sc-hKMtZM fjoSGl"
    >
      (product)
    </span>
  </div>
  <span
    class="sc-eCYdqJ haWGYV"
  >
    10%
  </span>
  <span
    class="sc-jSMfEi cbNCPN"
  >
    lender
  </span>
</div>
`;
